spring-cloud-gcp-data-spanner/src/main/java/org/springframework/cloud/gcp/data/spanner/repository/query/SpannerStatementQueryExecutor.java

```diff
@@ -182,18 +224,17 @@ public static String getColumnsStringForSelect(
 				.getPersistentProperty(o.getProperty()).getColumnName());
 		buildLimit(tree, stringBuilder);
 
-		stringBuilder.append(";");
-		return Pair.of(stringBuilder.toString(), tags);
+		String selectSql = stringBuilder.toString();
+		return Pair.of(tree.isExistsProjection() || tree.isCountProjection()
+				? "SELECT COUNT(1) FROM (" + selectSql + ")"
```

Comment:
Didn't we say during the review that this might have to be `COUNT(DISTINCT *)` or something? What was the result of that investigation?

Reply:
I don't think that works. If you were to say countTop100DistinctByColor that would mean you want to select the top 100 and count how many distinct there are in those 100. This means the answer is 100 or less. 

If you do not wrap a count around the entire statement like I do and instead do what you suggest
`SELECT COUNT(DISTINCT *) FROM TABLE WHERE COLOR = 'red' LIMIT 100` then I think that means you limit the number of final result rows to 100 (the count result), and not what is actually counted.

Reply:
I see, so you expect the user to use `DISTINCT` explicitly. OK.